packages/rest/src/lib/RequestManager.ts

```diff
@@ -203,6 +203,7 @@ export class RequestManager extends EventEmitter {
 		this.options = { ...DefaultRestOptions, ...options };
 		this.options.offset = Math.max(0, this.options.offset);
 		this.globalRemaining = this.options.globalRequestsPerSecond;
+		this.agent = options.agent ?? null;
```

Comment:
The DefaultRestOptions use `getGlobalDispatcher()` for the agent, then we check if the user has provided an agent via `RESTOptions` -> `options`, and if this isn't provided, we set it to `null`. So I guess this will collide here, no?

Reply:
`options` is the param passed to the constructor by the dev, it can have `agent` missing. Although the DefaultOptions's agent should be a getter so that it can update if a global dispatcher is set after loading REST...

Reply:
You're right that it was set to the global dispatcher, but I'm not entirely sure how to be honest. I think being able to set a dispatcher for the Rest manager instance will solve every use case though.

Reply:
Yeah I do think it solves every use case, but the expectation of people setting the global dispatcher and it not working seems like we need to fix it somehow down the line 🤔 
Like yes, it would probably route all traffic, even non-discord one through that, but if people expect that(?)

Reply:
Fixed this issue - had to update discord.js' undici so that the dispatcher was truly global! I was wondering why `getGlobalDispatcher` wasn't returning the ProxyAgent.